Fix exit codes in health check script

yehy4 opened the pull request:
Before this PR, if `ENABLE_IPv6` is set to `0` (which is [the default](https://github.com/padhi-homelab/docker_tang/blob/05d1164dba9ee589abd146bb7a86c0d2b5f79f8e/Dockerfile#L84)), then the health check script always exits with code `1` (reporting an unhealthy state) regardless of whether the IPv4 check succeeds or not. 

This happens because the check `[ "$ENABLE_IPv6" = "1" ]` is the last command that runs in the script (when `ENABLE_IPv6=0`), and since it evaluates to `false` (i.e., returns exit code `1`), the entire script exits with exit code `1`, indicating a failure.

In this PR, we explicitly use `if` statements (instead of command chaining using `&&`) to prevent this issue in `check-servers.sh`, and we use `if` statements in `start-servers.sh` to maintain consistency among all shell scripts in the repository.

SaswatPadhi commented:
Ah! This explains why the CI jobs on another repo that uses this tang image started failing.
I was being lazy and used a shorthand for `if-then-else` :sweat_smile: ... should have thought more carefully about the exit code.

SaswatPadhi commented on check-servers.sh:
```diff
@@ -1,4 +1,6 @@
 
-[ "$ENABLE_IPv4" = "1" ] && wget -4qSO /dev/null http://127.0.0.1:8080/adv
-[ "$ENABLE_IPv6" = "1" ] && wget -6qSO /dev/null http://[::1]:8080/adv
+[ "$ENABLE_IPv4" = "1" ] && $(wget -4qSO /dev/null http://127.0.0.1:8080/adv || exit 1)
```
I think it might be cleaner to use an `if-then-else` block instead of using subshell expansions and `exit`s.

```suggestion
if [ "$ENABLE_IPv4" = "1" ]; then
  wget -4qSO /dev/null http://127.0.0.1:8080/adv
fi
```

(same comment on the `ENABLE_IPv6` case below.

SaswatPadhi reviewed (changes requested):
I have one request on refactoring the conditional statement.

Also, we should make these fixes on the `start-servers.sh` script too.

yehy4 commented:
Done. I agree that this looks much cleaner now without the explicit `exit` calls. Thanks for the suggestion!

I added another commit to use `if` statements in `start-servers.sh` for consistency and updated the PR description to reflect the new changes.

SaswatPadhi reviewed (approved):
Looks great!

Thanks again :tada:
